datadog-crashtracker/src/crash_info/error_data.rs

```diff
@@ -16,24 +22,45 @@ pub struct ErrorData {
     pub threads: Vec<ThreadData>,
 }
 
+#[cfg(unix)]
+#[derive(Default)]
+pub struct CachedElfResolvers {
+    elf_resolvers: HashMap<PathBuf, ElfResolver>,
+}
+
+#[cfg(unix)]
+impl CachedElfResolvers {
+    pub fn get(&mut self, file_path: &PathBuf) -> anyhow::Result<&ElfResolver> {
+        use anyhow::Context;
+        if !self.elf_resolvers.contains_key(file_path.as_path()) {
```

Comment:
We could potentially use `Entry` here, which would save the double lookup but potentially require an extra allocation.